mlflow/models/model.py

```diff
@@ -651,10 +689,22 @@ def load(cls, path) -> "Model":
             )
 
         path = download_artifacts(artifact_uri=path)
+        env_vars = None
         if os.path.isdir(path):
             path = os.path.join(path, MLMODEL_FILE_NAME)
+            env_var_path = os.path.join(path, ENV_VAR_FILE_NAME)
+        elif os.path.isfile(path):
+            env_var_path = os.path.join(os.path.dirname(path), ENV_VAR_FILE_NAME)
+        else:
+            env_var_path = None
+        if os.path.exists(env_var_path):
+            # comments start with `#` such as ENV_VAR_FILE_HEADER
+            lines = Path(env_var_path).read_text().splitlines()
+            env_vars = [line for line in lines if line and not line.startswith("#")]
         with open(path) as f:
-            return cls.from_dict(yaml.safe_load(f.read()))
+            model_dict = yaml.safe_load(f.read())
+        model_dict["env_vars"] = env_vars
```

Comment:
```suggestion
        if env_var:
            model_dict["env_vars"] = env_vars
```

Reply:
Why do we need this? Passing None is also valid

Reply:
You're right. Never mind this comment.